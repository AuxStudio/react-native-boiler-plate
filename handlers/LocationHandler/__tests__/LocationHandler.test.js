import React from 'react';
import renderer from 'react-test-renderer';

import { LocationHandler } from '..';

describe('LocationHandler', () => {
  let spy;
  const dispatch = jest.fn();

  it('renders with all/minimum required props', () => {
    expect(renderer.create(<LocationHandler dispatch={jest.fn()} />)).toMatchSnapshot();
  });

  it('call getLocationPermission on componentDidMount', () => {
    spy = jest.spyOn(LocationHandler.prototype, 'getLocationPermission');

    renderer.create(<LocationHandler dispatch={dispatch} />);

    expect(spy).toHaveBeenCalled();
    expect(dispatch).toMatchSnapshot();
  });

  afterEach(() => {
    if (spy) {
      spy.mockClear();
    }
  });
});
