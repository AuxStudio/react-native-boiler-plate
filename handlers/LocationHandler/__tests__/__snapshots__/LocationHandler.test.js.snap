// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationHandler call getLocationPermission on componentDidMount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "getDeviceLocation",
          },
        },
        "payload": Object {
          "permission": "location",
        },
        "type": "checkAndRequestPermission",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`LocationHandler renders with all/minimum required props 1`] = `null`;
